networkx/classes/graph.py

```diff
@@ -852,14 +854,15 @@ def add_edges_from(self, ebunch, attr_dict=None, **attr):
         >>> G.add_edges_from([(3,4),(1,4)], label='WN2898')
         """
         # set up attribute dict
+        edge_attr = {}
         if attr_dict is None:
-            attr_dict = attr
-        else:
-            try:
-                attr_dict.update(attr)
-            except AttributeError:
+            attr_dict = {}
+        try:
+                edge_attr.update(attr_dict)
```

Comment:
Let's change this to:

    if attr_dict is not None:
        try:
            edge_attr.update(attr_dict)
        except TypeError:
            msg = "The attr_dict argument must be a dict or iterable of 2-tuples"
            raise NetworkXError(msg)

This removes the whole try/except portion when ```attr_dict is None``` (which happens a lot).